util/devel/test/apptainer/tryit.py

```diff
@@ -0,0 +1,156 @@
+    Prints a summary of the output at the end.
+    Saves all of the command output in a file called 'log'. """
+
+import argparse, glob, os, subprocess, shutil
+
+def gatherDirs(skip_nollvm):
+    dirs = [ ]
+    for d in glob.glob('current/*'):
+        if os.path.exists(os.path.join(d, 'image.def')):
+            if skip_nollvm and 'nollvm' in d:
+                pass # skip -nollvm versions if requested
+            else:
+                dirs.append(d)
+    dirs.sort()
+    return dirs
+
+def dirNameToConfigName(d):
+    return os.path.basename(d.rstrip('/'))
+
+def printAndLog(log, s):
+    print(s)
+    print(s, file=log)
+
+# Returns a tuple consisting of (exit code, last line)
+def runAndLog(log, command):
+    printAndLog(log, "Running command: {}\n".format(" ".join(command)))
+    try:
+        p = subprocess.Popen(command,
+                             stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                             encoding='utf-8')
+        lastline = ""
+        while p.poll() is None:
+            line = p.stdout.readline()
+            if line:
+                line = line.strip()
+                printAndLog(log, line)
+                if line and not line.startswith("INFO"):
+                    lastline = line
+        return (p.returncode, lastline)
+    except OSError:
+        return (-1, "")
+
+def main():
+    parser = argparse.ArgumentParser(
+            description='Run a command on all the images.')
+    parser.add_argument('--cleanup', dest='cleanup', action='store_true',
+                        help='delete each image and checkout after using it')
+    parser.add_argument('--rebuild', dest='rebuild', action='store_true',
+                        help='rebuild each image before running it')
+    parser.add_argument('--only', dest='only', action='store',
+                        help='only run one configuration')
+    parser.add_argument('--start', dest='start', action='store',
+                        help='run configurations starting with the passed one')
+    parser.add_argument('--skip-nollvm', dest='skip_nollvm', action='store_true',
+                        help='skip nollvm configurations')
+    parser.add_argument('command', nargs='+', help='command to run')
+
+    args = parser.parse_args()
+
+    logpath = 'log'
+
+    startDir = os.getcwd()
+
+    status = { }
+
+    with open(logpath, 'w', encoding="utf-8") as log:
+        dirs = gatherDirs(args.skip_nollvm)
+        if args.only != None:
+            hadDirs = dirs
+            dirs = [ ]
+            for d in hadDirs:
+                if dirNameToConfigName(d) == dirNameToConfigName(args.only):
+                    dirs.append(d)
+
+
+        if args.start != None:
+            hadDirs = dirs
+            dirs = [ ]
+            started = False
+            for d in hadDirs:
+                if dirNameToConfigName(d) == dirNameToConfigName(args.start):
+                    started = True
```

Comment:
Can these lines combined into one?

if dirNameToConfigName(d) == dirNameToConfigName(args.start):
dirs.append(d)

Or, is this because the dirs are sorted so once started is set to true, rest of the dirs that follws the started dir is added to dirs?

Reply:
> Or, is this because the dirs are sorted so once started is set to true, rest of the dirs that follws the started dir is added to dirs?

yes exactly